app/v1/controllers/validators/CreateAndAmendOtherDeductionsValidatorFactory.scala

```diff
@@ -0,0 +1,109 @@
+
+      def validate: Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] =
+        (
+          ResolveNino(nino),
+          ResolveTaxYear(appConfig.minimumPermittedTaxYear, taxYear, None, None),
+          resolveJson(body)
+        ).mapN(CreateAndAmendOtherDeductionsRequestData) andThen validateBodyFieldFormat
+
+    }
+
+  private def validateBodyFieldFormat(
+      parsed: CreateAndAmendOtherDeductionsRequestData): Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] = {
+
+    val validationResult = parsed.body.seafarers match {
+      case None => Valid(())
+      case Some(seafarers) =>
+        seafarers.zipWithIndex
+          .traverse_ { case (item, index) => validateSeafarers(item, index) }
+    }
+
+    validationResult.map(_ => parsed)
+  }
+
+  private def validateSeafarers(seafarers: Seafarers, arrayIndex: Int): Validated[Seq[MtdError], Unit] = {
+    import seafarers._
+
+    (
+      validateCustomerReference(customerReference, s"/seafarers/$arrayIndex/customerReference"),
+      validateAmountDeducted(amountDeducted, s"/seafarers/$arrayIndex/amountDeducted"),
+      validateNameOfShip(nameOfShip, s"/seafarers/$arrayIndex/nameOfShip"),
+      validateDates(fromDate, toDate, arrayIndex)
+    ).tupled
+      .andThen { case (_, _, _, _) => Validated.Valid(()) }
+  }
+
+  private def validateDates(fromDate: String, toDate: String, arrayIndex: Int): Validated[Seq[MtdError], Unit] = {
+    val fromPath = s"/seafarers/$arrayIndex/fromDate"
+    val toPath   = s"/seafarers/$arrayIndex/toDate"
+
+    object ResolveToFromDateRange extends DateRangeResolving {
+      override protected val startDateFormatError: MtdError = DateFormatError.withPath(fromPath)
+      override protected val endDateFormatError: MtdError   = DateFormatError.withPath(toPath)
+
+      def apply(value: (String, String), maybeError: Option[MtdError], path: Option[String]): Validated[Seq[MtdError], DateRange] = {
+        resolve(value, maybeError, path)
+      }
+
+    }
+
+    ResolveToFromDateRange((fromDate, toDate), Some(RangeToDateBeforeFromDateError.withPaths(List(fromPath, toPath))), None).map(_ => ())
+  }
+
+  private def validateCustomerReference(customerReference: Option[String], path: String): Validated[Seq[MtdError], Unit] =
+    customerReference match {
+      case None => Valid(())
+      case Some(ref: String) =>
+        if (customerRefRegex.matches(ref)) Valid(()) else Invalid(Seq(CustomerReferenceFormatError.withPath(path)))
+    }
+
+  private val resolveAmountDeducted = ResolveParsedNumber()
+
+  private def validateAmountDeducted(value: BigDecimal, path: String): Validated[Seq[MtdError], Unit] = {
+    resolveAmountDeducted(value, path = Some(path)).map(_ => ())
+  }
+
+  private def validateNameOfShip(field: String, path: String): Validated[Seq[MtdError], Unit] = {
+    if (field.length <= 105) Valid(()) else Invalid(List(NameOfShipFormatError.copy(paths = Some(Seq(path)))))
```

Comment:
has got "List(.." and "Seq(.." styles on the same line, can they both  be List(..)?

Reply:
actually ought to use ".withPath(path)" instead, solves the issue then